src/resource.js

```diff
@@ -71,12 +69,36 @@ export default class Resource {
    * @returns {String|Array|Object}
    */
   get source() {
-    if (this._sourceKey === 'path' && this._basePath) {
-      const resourcePath = this.descriptor[this._sourceKey]
-      return path.normalize(`${this._basePath}/${resourcePath}`)
-    }
+    const source = this.descriptor[this._sourceKey]
+
+    if (!source) {
+      // Neither inline data nor path available
+      return null
+
+    } else if (this.typeOfResourcePath === 'inline') {
+      // Data is inline
+      return source
 
-    return this.descriptor[this._sourceKey]
+    } else if (this._sourceKey === 'path' && this._basePath === '') {
+      // Local path, no basePath provided
```

Comment:
Not sure why `this.__sourceKey === 'path'` means that a local path:

```javascript
  get _sourceKey() {
    const inlineData = this.descriptor.data

    if (inlineData) {
      return 'data'
    }

    return 'path'
  }
```

Based on this code remote path also satisfies this condition.